tests/test_module_loader.py

```diff
@@ -257,3 +257,46 @@ def test_load_module_v1_and_v2_installed(
     assert f"Module {module_name} is installed as a V1 module and a V2 module: V1 will be ignored." in caplog.text
     assert module_name in project.modules
     assert isinstance(project.modules[module_name], ModuleV2)
+
+
+def test_load_module_recursive_v2_module_depends_on_v1(local_module_package_index: str, snippetcompiler) -> None:
+    """
+    A V2 module cannot depend on a V1 module. This test case ensure that the load_module_recursive() method
+    raises an error when a dependency of a V2 module is only available as a V1 module.
+
+    Dependency graph:  v2_depends_on_v1 (V2)  --->  mod1 (V1)
+    """
+    project = snippetcompiler.setup_for_snippet(
+        snippet="import v2_depends_on_v1", python_package_source=local_module_package_index
+    )
+    with pytest.raises(ModuleLoadingException, match="could not find module mod1"):
+        project.load_module_recursive(install=True)
```

Comment:
Could you also add a variation where the v1 module already exists in `project.modules`? I think `project.get_module("mod1", allow_v1=True)` should do it. From what I discovered during work on my PR I expect this might fail because `get_module` doesn't check the constraint when it fetches something from the cache.

Reply:
You are right. This is a bug. I will fix this.

Reply:
I need to fix this as well for my PR so we'll have a conflict. It should be a simple fix of a few lines though so I propose we just resolve it when it arises.